### ISSUE TITLE
informative-docs doesn't work on exported declarations

### ISSUE THREAD

**Issue by sandersn** [gajus/eslint-plugin-jsdoc#1198]
## Expected behavior
Error on comment: "This description only repeats the name it describes"
## Actual behavior
No error.

Deleting the export keyword provides the correct error.
## ESLint Config

```js
// Format JS (or JSON) code here
```json
{
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "warnOnUnsupportedTypeScriptVersion": false,
        "ecmaVersion": 6,
        "sourceType": "module"
    },
    "env": {
        "browser": false,
        "node": true,
        "es6": true
    },
    "plugins": [
        "jsdoc"
    ],
    "rules": {
        "jsdoc/informative-docs": "error"
    }
}
```

## ESLint sample

Note: this is typescript but I removed the types so it should parse the same as JS.

```ts
/**
 * package name from path
 */
export function packageNameFromPath(path) {
  const base = basename(path);
  return /^v\d+(\.\d+)?$/.exec(base) || /^ts\d\.\d/.exec(base) ? basename(dirname(path)) : base;
}
```

## Environment

- Node version: 20.11.0
- ESLint version 8.56.0
- `eslint-plugin-jsdoc` version: 48.0.5
